app/views/declaration/fiscalInformation/additional_fiscal_references.scala.html

```diff
@@ -0,0 +1,136 @@
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import views.{BackButton, Title}
+@import views.html.components.gds._
+@import play.twirl.api.HtmlFormat
+@import scala.collection.immutable
+@import utils.ListItem._
+@import views.ErrorMapper.yesNoErrors
+@import config.AppConfig
+
+@this(
+        govukLayout: gdsMainTemplate,
+        govukFieldset: GovukFieldset,
+        govukButton: GovukButton,
+        govukTable : GovukTable,
+        govukRadios: GovukRadios,
+        errorSummary: errorSummary,
+        sectionHeader: sectionHeader,
+        pageTitle: pageTitle,
+        exportsInputText: exportsInputText,
+        spanVisuallyHidden: spanVisuallyHidden,
+        tariffDetails: tariffDetails,
+        addButton: addButton,
+        addAnotherQuestion: addAnotherQuestion,
+        link: link,
+        saveButtons: saveButtons,
+        formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+)
+
+@(mode: Mode, itemId:String, form: Form[YesNoAnswer], additionalReferences: Seq[AdditionalFiscalReference])(implicit request: JourneyRequest[_], messages: Messages)
+
+@title = @{
+    if(additionalReferences.size == 1)
+        Title("declaration.additionalFiscalReferences.table.heading", "declaration.section.5")
+    else
+        Title("declaration.additionalFiscalReferences.table.multiple.heading", "declaration.section.5", additionalReferences.size.toString)
+}
+
+@header = @{
+    if(additionalReferences.size == 1)
+        Text(messages("declaration.additionalFiscalReferences.table.heading"))
+    else
+        Text(messages("declaration.additionalFiscalReferences.table.multiple.heading", additionalReferences.size))
+}
+
+@referenceTable = {
+    @if(additionalReferences.nonEmpty) {
+        @govukTable(Table(
+            attributes = Map("id" -> "additional_fiscal_references"),
+            rows = additionalReferences.zipWithIndex.map { case (additionalReference, index) =>
+                Seq(
+                    TableRow(
+                        content = Text(additionalReference.countryName),
+                        attributes = Map("id" -> s"additional_fiscal_references-row-$index-country")
+                    ),
+                    TableRow(
+                        content = Text(additionalReference.reference),
+                        attributes = Map("id" -> s"additional_fiscal_references-row-$index-reference")
+                    ),
+                    TableRow(
+                        content = HtmlContent(link(
+                            messages("site.remove"),
+                            Some(messages("declaration.additionalInformation.table.remove.hint", additionalReference.reference)),
+                            AdditionalFiscalReferencesRemoveController.displayPage(mode, itemId, createId(index, additionalReference)))),
+                        classes = "govuk-table__cell--numeric",
+                        attributes = Map("id" -> s"additional_fiscal_references-row$index-remove_button")
```

Comment:
In the last `TableRow` the id is set to have `row$index` while in the others it is `row-$index`